src/create.cpp

```diff
@@ -40,25 +41,49 @@ unsigned Create::run()
         throw rest::Error(ret.error());
     }
 
-    std::string assetJson = m_request.body();
-
-    auto ret = AssetManager::createAsset(assetJson, user.login());
-    if (!ret) {
-        auditError(ret.error());
-        throw rest::errors::Internal(ret.error());
+    cxxtools::SerializationInfo si;
+    try {
+        std::stringstream          jsonIn(m_request.body());
+        cxxtools::JsonDeserializer deserializer(jsonIn);
+        deserializer.deserialize(si);
+    } catch (const std::exception& e) {
+        logError(e.what());
+        throw rest::errors::Internal(e.what());
     }
 
-    auto createdAsset = AssetManager::getItem(*ret);
-
-    if (!createdAsset) {
-        auditError(createdAsset.error());
-        throw rest::errors::Internal(createdAsset.error());        
-    } 
+    cxxtools::SerializationInfo assetsJsonList;
+    if (si.findMember("assets")) {
+        assetsJsonList = si.getMember("assets");
+    } else {
+        assetsJsonList.addMember(m_request.body());
+    }
 
-    m_reply << "{ " << utils::json::jsonify("id", createdAsset->name) << " }\n\n";
+    bool             someOk = false;
+    pack::StringList createdName;
+    for (const auto& it : assetsJsonList) {
+        auto ret = AssetManager::createAsset(it, user.login());
+        if (!ret) {
+            auditError(ret.error());
+            continue;
+        }
+
+        auto createdAsset = fty::asset::db::idToNameExtName(*ret);
+
+        if (!createdAsset) {
+            auditError(createdAsset.error());
+            continue;
+        }
+
+        createdName.append(createdAsset->first.c_str());
+        auditInfo("Request CREATE asset id {} SUCCESS"_tr, createdAsset->first.c_str());
+        someOk = true;
+    }
 
-    auditInfo("Request CREATE asset id {} SUCCESS"_tr, createdAsset->name);
+    m_reply << *pack::json::serialize(createdName) << "\n\n";
 
+    if (!someOk) {
+        throw rest::errors::Internal("Some of assets creation FAILED");
+    }
```

Comment:
if not someOk, then list is empty, right? why are you serialize it to output? Exists any reason?